lib/oauth2/access_token.rb

```diff
@@ -10,6 +10,7 @@ class << self
       # @param [Hash] a hash of AccessToken property values
       # @return [AccessToken] the initalized AccessToken
       def from_hash(client, hash)
+        hash = hash.dup
```

Comment:
This one seems redundant, looks like the `dup` added to initialize will already get this.